common/Scripts/ModifyWindowsOptionalFeatures.cs

```diff
@@ -13,105 +13,121 @@
 
 namespace DevHome.Common.Scripts;
 
-public static class ModifyWindowsOptionalFeatures
-{
-    public static async Task<ExitCode> ModifyFeaturesAsync(
+public class ModifyWindowsOptionalFeatures : IDisposable
+{
+    private readonly Process _process;
+    private readonly ILogger? _log;
+    private readonly CancellationToken _cancellationToken;
+    private readonly string _featuresString;
+    private Stopwatch _stopwatch = new();
+    private bool _disposed;
+
+    public ModifyWindowsOptionalFeatures(
         IEnumerable<WindowsOptionalFeatureState> features,
-        ILogger? log = null,
-        CancellationToken cancellationToken = default)
-    {
-        if (!features.Any(f => f.HasChanged))
-        {
-            return ExitCode.Success;
-        }
+        ILogger? log,
+        CancellationToken cancellationToken)
+    {
+        _log = log;
+        _cancellationToken = cancellationToken;
 
         // Format the argument for the PowerShell script using `n as a newline character since the list
         // will be parsed with ConvertFrom-StringData.
         // The format is FeatureName1=True|False`nFeatureName2=True|False`n...
-        var featuresString = string.Empty;
+        _featuresString = string.Empty;
         foreach (var featureState in features)
         {
             if (featureState.HasChanged)
             {
-                featuresString += $"{featureState.Feature.FeatureName}={featureState.IsEnabled}`n";
+                _featuresString += $"{featureState.Feature.FeatureName}={featureState.IsEnabled}`n";
```

Comment:
NIT: change _featuresString into a StringBuilder for string concatenation.

Left as a NIT because I don't believe the number of features will expand to the point that string concatenation will have a negative performance impact.

Reply:
Will leave as-is for now to avoid too much churn. I agree that we're bounded by the number of features we expose in the UI and it's a very small number at present. Good callout for future optimization though.